docs/_ext/scylladb_cc_properties.py

```diff
@@ -0,0 +1,46 @@
+import yaml
+import re
+from sphinx.application import Sphinx
+
+def create_yaml_file(destination, data):
+    current_data = None
+    
+    try:
+        with open(destination, 'r') as file:
+            current_data = yaml.safe_load(file)
+    except FileNotFoundError:
+        pass
+
+    if current_data != data:
+        with open(destination, 'w') as file:
+            yaml.dump(data, file)
+
+def parse_db_properties(config_file):
+    with open(config_file, 'r') as file:
+        content = file.read()
+
+    pattern = r',\s*(\w+)\(this,\s*"(\w+)",\s*(?:liveness::(\w+),\s*)?value_status::(\w+),\s*([^,]+),\s*"([^"]+)"\)'
+    matches = re.findall(pattern, content)
+
+    properties = []
+    for match in matches:
+        property_data = {
+            "name": match[1].strip(),
+            "value_status": match[3].strip(), 
+            "default": match[4].strip(),
+            "liveness": 'True' if match[2] else 'False',
+            "description": match[5].strip().replace('\n', '<br />')
+        }
+        properties.append(property_data)
+
+    return properties
+
+def generate_cc_docs(app: Sphinx):
+    config_file_path = '../db/config.cc'
+    destination_path = '_data/db_config.yaml'
+    parsed_properties = parse_db_properties(config_file_path)
+    
+    create_yaml_file(destination_path, parsed_properties)
+
+def setup(app: Sphinx):
+    app.connect("builder-inited", generate_cc_docs)
```

Comment:
might want to add the .cc file as a new dependency of current document, see https://www.sphinx-doc.org/en/master/extdev/envapi.html#sphinx.environment.BuildEnvironment.note_dependency, so that sphinx can rebuilt the document after the .cc file is updated.

Reply:
Added in https://github.com/scylladb/scylladb/pull/14973/commits/c9adf326a1ca6e0a00b02cbb2e02bb0ed3948bee